rbi/core/refinement.rbi

```diff
@@ -0,0 +1,54 @@
+# typed: __STDLIB_INTERNAL
+
+# [Refinement](https://docs.ruby-lang.org/en/3.2/Refinement.html) is a class of
+# the self (current context) inside refine statement. It allows to import
+# methods from other modules, see
+# [import_methods](https://docs.ruby-lang.org/en/3.2/Refinement.html#method-i-import_methods).
+class Refinement < Object
+  # Imports methods from modules. Unlike
+  # [Module#include](https://docs.ruby-lang.org/en/3.2/Module.html#method-i-include),
+  # [Refinement#import_methods](https://docs.ruby-lang.org/en/3.2/Refinement.html#method-i-import_methods)
+  # copies methods and adds them into the refinement, so the refinement is
+  # activated in the imported methods.
+  #
+  # Note that due to method copying, only methods defined in Ruby code can be imported.
+  #
+  # ```ruby
+  # module StrUtils
+  #   def indent(level)
+  #     ' ' * level + self
+  #   end
+  # end
+  #
+  # module M
+  #   refine String do
+  #     import_methods StrUtils
+  #   end
+  # end
+  #
+  # using M
+  # "foo".indent(3)
+  # #=> "   foo"
+  #
+  # module M
+  #   refine String do
+  #     import_methods Enumerable
+  #     # Can't import method which is not defined with Ruby code: Enumerable#drop
+  #   end
+  # end
+  # ```
+  #
+  # Also aliased as: import_methods
+  sig do
+    params(
+      module: Module,
+      rest: T::Array[Module],
+    )
+    .returns(Refinement)
+  end
+  def import_methods(module, *rest); end
```

Comment:
[Docs for this method](https://docs.ruby-lang.org/en/3.2/Refinement.html#method-i-import_methods) have `import_methods(module, ...) → self` as the signature, stating that it acts on "modules". I've made the signature reflect a 1+ approach with rest, but wanted to make sure this is how this would normally be approached.